src/Agent.Listener/Configuration/AgentConfigSettings.cs

```diff
@@ -0,0 +1,27 @@
+﻿
+namespace Microsoft.VisualStudio.Services.Agent.Listener.Configuration
+{
+
+    public class AgentConfigSettings
+    {
+        public int PoolId;
+
+        public AgentConfigSettings(int poolId)
+        {
+            this.PoolId = poolId;
+        }
+    }
+
+    public class DeploymentAgentConfigSettings : AgentConfigSettings
+    {
+        public int DeploymentGroupId;
+
+        public string ProjectId;
+
+        public DeploymentAgentConfigSettings(int poolId, int deploymentGroupID, string projectId) : base(poolId)
```

Comment:
we dont want any consumer consume it for any other state. Intention is to use these with this state only. If new state is required in future will add new contractor